parallel_algorithms/P3179/P3179.bs

```diff
@@ -713,14 +727,2242 @@ Here are questions we want to answer (potentially not a complete list):
+         class Proj = identity, class T = projected_value_t<I, Proj>>
+  requires indirectly_comparable<I, const T*, Pred, Proj>
+  subrange<I> ranges::search_n(ExecutionPolicy&& exec, I first, S last, iter_difference_t<I> count, const T& value, Pred pred = {}, Proj proj = {});
+
+template<class ExecutionPolicy, random_access_range R, class Pred = ranges::equal_to, class Proj = identity, class T = projected_value_t<iterator_t<R>, Proj>>
+  requires indirectly_comparable<iterator_t<R>, const T*, Pred, Proj> && sized_range<R>
+  borrowed_subrange_t<R> ranges::search_n(ExecutionPolicy&& exec, R&& r, range_difference_t<R> count, const T& value, Pred pred = {}, Proj proj = {});
+```
+
+## Modify `starts_with` in <a href="https://eel.is/c++draft/alg.starts.with">[**alg.starts.with**]</a> ## {#modify_alg_starts_with}
+
+<xmp>
+template<input_iterator I1, sentinel_for<I1> S1, input_iterator I2, sentinel_for<I2> S2,
+         class Pred = ranges::equal_to, class Proj1 = identity, class Proj2 = identity>
+  requires indirectly_comparable<I1, I2, Pred, Proj1, Proj2>
+  constexpr bool ranges::starts_with(I1 first1, S1 last1, I2 first2, S2 last2, Pred pred = {},
+                                     Proj1 proj1 = {}, Proj2 proj2 = {});
+template<input_range R1, input_range R2, class Pred = ranges::equal_to, class Proj1 = identity,
+         class Proj2 = identity>
+  requires indirectly_comparable<iterator_t<R1>, iterator_t<R2>, Pred, Proj1, Proj2>
+  constexpr bool ranges::starts_with(R1&& r1, R2&& r2, Pred pred = {},
+                                     Proj1 proj1 = {}, Proj2 proj2 = {});
+</xmp>
+
+<i>Returns</i>: `ranges::mismatch(std::move(first1), last1, std::move(first2), last2,
+                   pred, proj1, proj2).in2 == last2`
+
+<ins>
+<xmp class="wordingStyle">
+template<class ExecutionPolicy, random_access_iterator I1, sentinel_for<I1> S1, random_access_iterator I2, sentinel_for<I2> S2,
+         class Pred = ranges::equal_to, class Proj1 = identity, class Proj2 = identity>
+  requires indirectly_comparable<I1, I2, Pred, Proj1, Proj2> && (sized_sentinel_for<S1, I1> || sized_sentinel_for<S2, I2>)
+  bool ranges::starts_with(ExecutionPolicy&& exec, I1 first1, S1 last1, I2 first2, S2 last2,
+                           Pred pred = {}, Proj1 proj1 = {}, Proj2 proj2 = {});
+template<class ExecutionPolicy, random_access_range R1, random_access_range R2, class Pred = ranges::equal_to,
+         class Proj1 = identity, class Proj2 = identity>
+  requires indirectly_comparable<iterator_t<R1>, iterator_t<R2>, Pred, Proj1, Proj2> && (sized_range<R1> || sized_range<R2>)
+  bool ranges::starts_with(ExecutionPolicy&& exec, R1&& r1, R2&& r2, Pred pred = {}, Proj1 proj1 = {}, Proj2 proj2 = {});
+</xmp>
+
+<i>Returns</i>:
+    `ranges::mismatch(std::forward<ExecutionPolicy>(exec), std::move(first1), last1, std::move(first2), last2,
+                          pred, proj1, proj2).in2 == last2`
```

Comment:
I would double-check that use of `||` for `sized_range` requirements is right for this algorithm. The concern is about `mismatch(...).in2==last2` for the return value. If the second range `[first2,last2)` is unbounded, will the returned iterator to the range ever be equal to `last2`, even if all elements of the first range were matched?

Or in other words, does not the semantics of `starts_with` assume that the second sequence is finite?

Reply:
I think we agreed on "both sized"